dbt/include/hive/macros/adapters.sql

```diff
@@ -258,4 +257,20 @@
   {% endcall %}
   {% do return(load_result('list_views_without_caching').table) %}
 {% endmacro %}
-k
+
+{% macro get_hive_version() %}
+  {% set version_results = run_query('select version()') %}
+  {% if execute %}
+     {% set num_rows = (version_results | length) %}
+
+     {% if num_rows == 1 %}
+        {% set version_text = version_results[0][0] %}
+        {{ log("get_hive_version " ~ version_text) }}
+        {% do return(version_text.split(".")[0]) %}
+     {% else %}
+        return '2';  {# assume hive 2 by default #}    
```

Comment:
is L270-272 redundant

Reply:
I don't think it is redundant, the macro containing query is typically parsed and then executed, so we need to handle the executed state separately (that is when the actual query results) are available as against parsing state. Ref: https://docs.getdbt.com/reference/dbt-jinja-functions/execute.
There was a typo though the in the return statement that I have fixed. 

Also attaching screenshots of Hive 2 vs 3 runs for incremental model with merge for reference.
![image](https://user-images.githubusercontent.com/4781546/189050933-cb961410-2a6b-4239-a532-c06e0a8f327b.png)
<img width="1243" alt="image" src="https://user-images.githubusercontent.com/4781546/189062681-2b4762d4-3f04-436a-ab93-9bf3e38149cc.png">